homeassistant/components/epic_games_store/__init__.py

```diff
@@ -0,0 +1,36 @@
+"""The Epic Games Store integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import EGSCalendarUpdateCoordinator
+
+PLATFORMS: list[Platform] = [
+    Platform.CALENDAR,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Epic Games Store from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    coordinator = EGSCalendarUpdateCoordinator(hass, entry)
```

Comment:
Can we create the `EpicGamesStoreAPI` before this and check it works? If we can catch an exception and raise `ConfigEntryNotReady` it's a bit more user friendly.

We can then insert the object into the coordinator

Reply:
Was like that before 622440e but I found the current solution much more clean and avoid code duplication

`coordinator.async_config_entry_first_refresh()` is raising `ConfigEntryNotReady`, with an even better error handling:

https://github.com/home-assistant/core/blob/63ef9efa26f97c31b0915753e0f1fd83abdf49a2/homeassistant/helpers/update_coordinator.py#L258-L272


Don't worry, other integrations are doing it like that.
![image](https://github.com/home-assistant/core/assets/14821482/2029ce1e-cee7-48bd-9cd0-c879c306cf5c)